### ISSUE TITLE
[Paperless-NGX] error: can't build this add-on!

### ISSUE THREAD

**Issue by pingme01** [alexbelgium/hassio-addons#275]
- Addon name : Paperless-NGX
- Addon version :ngx-1.6.0-ls1ngx-1.6.0-ls1

When I want to update from version 1.5.0-8  to 1.6.0-ls1ngx-1.6.0-ls1, I get the message: Invalid build environment, can't build this add-on!

When I completely remove the addon, I can't install the new addon either. I get the same error in Supervisor.

Version
core-2022.3.7

Supervisor
Version
supervisor-2022.03.5

Operating System
Home Assistant OS 7.5

Used space
49.1 %

**Comment by alexbelgium** [comment]
Hi, complex issue as I can't reproduce...
It's an upgrade from paperless-ng to paperless-ngx which is a big change, but as it builds on my system and would have expected as such on yours.

Any additional infos in the supervisor log ?

Thanks!

**Comment by pingme01** [comment]
Home-Assistant has a static IP assigned.
Just upgraded to 2022.3.8, but the issue remains.
Here is some additional information from the supervisor log:

22-03-29 19:13:52 INFO (MainThread) [supervisor.backups.manager] Creating new partial backup with slug 1e602c22
22-03-29 19:13:52 INFO (MainThread) [supervisor.backups.manager] Backing up 1e602c22 store Add-ons
22-03-29 19:14:42 INFO (SyncWorker_3) [supervisor.docker.addon] Export image db21ed7f/amd64-addon-paperless_ng to /data/tmp/tmph28sjz8r/image.tar
22-03-29 19:14:50 INFO (SyncWorker_3) [supervisor.docker.addon] Export image db21ed7f/amd64-addon-paperless_ng done
22-03-29 19:14:50 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on db21ed7f_paperless_ng
22-03-29 19:20:46 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon db21ed7f_paperless_ng
22-03-29 19:20:48 INFO (MainThread) [supervisor.backups.manager] Creating partial backup with slug 1e602c22 completed
22-03-29 19:20:48 INFO (SyncWorker_3) [supervisor.docker.interface] Updating image db21ed7f/amd64-addon-paperless_ng:1.5.0-8 to db21ed7f/amd64-addon-paperless_ng:ngx-1.6.0-ls1
22-03-29 19:20:48 ERROR (SyncWorker_3) [supervisor.docker.addon] Invalid build environment, can't build this add-on!

That's all there is...is there anything else that I can provide?

**Comment by alexbelgium** [comment]
Mmh... I'm really stuck here as I can't replicate :) 

Dio you use other locally build addons from my repo without issues?

And in the past did it worked to install the previous versions?

What is your arch (aarch64, armv7, amd64)?

**Comment by alexbelgium** [comment]
BTW, I suppose "Addon version :ngx-1.6.0-ls1ngx-1.6.0-ls1" is a typo? That the addon version is just "ngx-1.6.0-ls1"? Thanks:)

**Comment by pingme01** [comment]
sorry that was a typo indeed! This is what it says when I want to do the upgrade:

_Update Paperless NG
You have 1.5.0-8 installed. Click update to update to version ngx-1.6.0-ls1_

I've installed Photoprism from your repo and that installs without any issues. As many of the other addons that you've created! Big thanks for that! 

I've also tried to remove the current addon and do a clean install, but that failed as well.
Should I completely remove the 1.5.0-8 addon and check if everything is gone with your Portainer addon?

**Comment by alexbelgium** [comment]
Well I really don't understand the issue... You could indeed check in portainer if there are no residues from other images that corms interfer. I'll check tomorrow if there is no hidden crlf in my scripts. What is strange is that it builds fine here :)

I've got a second system on amd64, I'll try to build on that one tomorrow

**Comment by alexbelgium** [comment]
Great news, I can replicate it on amd64 :-) So the issue is only with amd64 builds

**Comment by alexbelgium** [comment]
ah ah, I had the issue (invalid build environement) on amd64, but clicked several times on install ; it seemed to fail ; I restarted HA and it installed fine ;-)

**Comment by alexbelgium** [comment]
And now that it has installed once, I can't repeat the error message... Each builds now successes even on amd64. So perhaps you could try that? Full update, reboot HA, try installing the addon several times ? Sorry for that lack of specificities but that was what made the error message disappear ;-)

**Comment by pingme01** [comment]
Rebooted and tried to installed it multiple times, but unfortunately no luck....

After uninstalling the addon, there was an image left, which I deleted using Portainer. 
After that, the new version installed. Unfortunately I have to do the uploading of documents again, but that's not a big deal (at least for me ;-)). Still weird that the old image, is blocking the installation of the new version.

**Comment by alexbelgium** [comment]
that's indeed strange... glad it worked out in the end! 
Regarding data loss, it all depends on the path you used for files. If you used the default "PAPERLESS_DATA_DIR=/config/addons_config/paperless_ng" hopefully all should still be in this directory ?

**Comment by pingme01** [comment]
I think I made an user error with the documents.... I logged in as admin and did see the documents, however previews weren't working. Therefor I thought that it was just a thumbnail, so immediately deleted everything. I didn't have huge amount of documents added, since I was still playing around with the tool...

Thank you very much for your support and all the excellent addons! Keep up the good work!!
